From 68aa09d58dd62b01f121a4f2eedac4a7c28d4f15 Mon Sep 17 00:00:00 2001
From: Mat Booth
Date: Wed, 4 May 2016 01:26:19 +0100
Subject: Bug 396457 - [Releng][Maven] Create a build based on tycho

First part of adding the necessary machinery to build ECF with tycho.
Adds two new projects, one to contain target platform definitions,
and one to contain p2 repository definitions. Adds a parent pom at
the root of the project and a gitignore file to mask built artifacts
from the git repo. Poms are added and the build enabled for all the
core features and plug-ins used by Equinox P2 (i.e. the minimum
necessary to build the Eclipse Platform itself.)

Change-Id: I0b41b4a611e2b3588fe876f14bd7c8e80de07b67
Signed-off-by: Mat Booth <mat.booth@redhat.com>
---
 .gitignore                                         |   1 +
 .../bundles/org.eclipse.ecf.filetransfer/pom.xml   |  15 ++
 framework/bundles/org.eclipse.ecf.identity/pom.xml |  15 ++
 framework/bundles/org.eclipse.ecf.ssl/pom.xml      |  15 ++
 framework/bundles/org.eclipse.ecf/pom.xml          |  15 ++
 pom.xml                                            | 284 +++++++++++++++++++++
 .../pom.xml                                        |  15 ++
 .../pom.xml                                        |  15 ++
 .../pom.xml                                        |  15 ++
 .../org.eclipse.ecf.provider.filetransfer/pom.xml  |  15 ++
 .../features/org.eclipse.ecf.core.feature/pom.xml  |  15 ++
 .../org.eclipse.ecf.core.ssl.feature/pom.xml       |  15 ++
 .../org.eclipse.ecf.filetransfer.feature/pom.xml   |  15 ++
 .../pom.xml                                        |  15 ++
 .../pom.xml                                        |  15 ++
 .../pom.xml                                        |  15 ++
 releng/org.eclipse.ecf.releng.repository/.project  |  11 +
 .../org.eclipse.ecf.releng.repository/category.xml |  22 ++
 releng/org.eclipse.ecf.releng.repository/pom.xml   |  16 ++
 releng/org.eclipse.ecf.releng.target/.project      |  11 +
 .../org.eclipse.ecf.releng.target/ecf-luna.target  |  38 +++
 .../org.eclipse.ecf.releng.target/ecf-mars.target  |  38 +++
 .../org.eclipse.ecf.releng.target/ecf-neon.target  |  41 +++
 releng/org.eclipse.ecf.releng.target/pom.xml       |  55 ++++
 24 files changed, 727 insertions(+)
 create mode 100644 .gitignore
 create mode 100644 framework/bundles/org.eclipse.ecf.filetransfer/pom.xml
 create mode 100644 framework/bundles/org.eclipse.ecf.identity/pom.xml
 create mode 100644 framework/bundles/org.eclipse.ecf.ssl/pom.xml
 create mode 100644 framework/bundles/org.eclipse.ecf/pom.xml
 create mode 100644 pom.xml
 create mode 100644 providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl/pom.xml
 create mode 100644 providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4/pom.xml
 create mode 100644 providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl/pom.xml
 create mode 100644 providers/bundles/org.eclipse.ecf.provider.filetransfer/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.core.feature/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.core.ssl.feature/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.filetransfer.feature/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature/pom.xml
 create mode 100644 releng/features/org.eclipse.ecf.filetransfer.ssl.feature/pom.xml
 create mode 100644 releng/org.eclipse.ecf.releng.repository/.project
 create mode 100644 releng/org.eclipse.ecf.releng.repository/category.xml
 create mode 100644 releng/org.eclipse.ecf.releng.repository/pom.xml
 create mode 100644 releng/org.eclipse.ecf.releng.target/.project
 create mode 100644 releng/org.eclipse.ecf.releng.target/ecf-luna.target
 create mode 100644 releng/org.eclipse.ecf.releng.target/ecf-mars.target
 create mode 100644 releng/org.eclipse.ecf.releng.target/ecf-neon.target
 create mode 100644 releng/org.eclipse.ecf.releng.target/pom.xml

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..2f7896d
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1 @@
+target/
diff --git a/framework/bundles/org.eclipse.ecf.filetransfer/pom.xml b/framework/bundles/org.eclipse.ecf.filetransfer/pom.xml
new file mode 100644
index 0000000..62854c1
--- /dev/null
+++ b/framework/bundles/org.eclipse.ecf.filetransfer/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.filetransfer</artifactId>
+  <version>5.0.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/framework/bundles/org.eclipse.ecf.identity/pom.xml b/framework/bundles/org.eclipse.ecf.identity/pom.xml
new file mode 100644
index 0000000..29415e5
--- /dev/null
+++ b/framework/bundles/org.eclipse.ecf.identity/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.identity</artifactId>
+  <version>3.7.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/framework/bundles/org.eclipse.ecf.ssl/pom.xml b/framework/bundles/org.eclipse.ecf.ssl/pom.xml
new file mode 100644
index 0000000..645f1a2
--- /dev/null
+++ b/framework/bundles/org.eclipse.ecf.ssl/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.ssl</artifactId>
+  <version>1.2.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/framework/bundles/org.eclipse.ecf/pom.xml b/framework/bundles/org.eclipse.ecf/pom.xml
new file mode 100644
index 0000000..3cbd474
--- /dev/null
+++ b/framework/bundles/org.eclipse.ecf/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf</artifactId>
+  <version>3.8.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/pom.xml b/pom.xml
new file mode 100644
index 0000000..066d87b
--- /dev/null
+++ b/pom.xml
@@ -0,0 +1,284 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+        Copyright (C) 2016, Red Hat, Inc
+
+   All rights reserved. This program and the accompanying materials
+   are made available under the terms of the Eclipse Public License v1.0
+   which accompanies this distribution, and is available at
+   http://www.eclipse.org/legal/epl-v10.html
+-->
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>ecf-parent</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+
+  <name>Eclipse Communication Framework</name>
+  <url>https://www.eclipse.org/ecf/</url>
+
+  <licenses>
+    <license>
+      <name>Eclipse Public License v1.0</name>
+      <url>http://www.eclipse.org/legal/epl-v10.html</url>
+      <distribution>repo</distribution>
+      <comments>
+        All rights reserved.
+
+        This program and the accompanying materials are made
+        available under the terms of the Eclipse Public License v1.0
+        which accompanies this distribution, and is available at
+        http://www.eclipse.org/legal/epl-v10.htm
+      </comments>
+    </license>
+  </licenses>
+
+  <issueManagement>
+    <url>https://bugs.eclipse.org/bugs/buglist.cgi?product=ECF</url>
+    <system>Bugzilla</system>
+  </issueManagement>
+
+  <properties>
+    <tycho-version>0.25.0</tycho-version>
+    <cbi-version>1.1.3</cbi-version>
+    <target-platform>neon</target-platform>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.resources.sourceEncoding>UTF-8</project.resources.sourceEncoding>
+    <tycho.scmUrl>scm:git:https://git.eclipse.org/r/ecf/org.eclipse.ecf</tycho.scmUrl>
+    <!-- TODO: tests are not yet ready for tycho -->
+    <skipTests>true</skipTests>
+  </properties>
+
+  <pluginRepositories>
+    <pluginRepository>
+      <id>repo.eclipse.org</id>
+      <url>https://repo.eclipse.org/content/groups/releases/</url>
+    </pluginRepository>
+  </pluginRepositories>
+
+  <modules>
+    <module>framework/bundles/org.eclipse.ecf</module>
+    <module>framework/bundles/org.eclipse.ecf.filetransfer</module>
+    <module>framework/bundles/org.eclipse.ecf.identity</module>
+    <module>framework/bundles/org.eclipse.ecf.ssl</module>
+
+    <module>providers/bundles/org.eclipse.ecf.provider.filetransfer</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl</module>
+
+    <module>releng/features/org.eclipse.ecf.core.feature</module>
+    <module>releng/features/org.eclipse.ecf.core.ssl.feature</module>
+    <module>releng/features/org.eclipse.ecf.filetransfer.feature</module>
+    <module>releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature</module>
+    <module>releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature</module>
+    <module>releng/features/org.eclipse.ecf.filetransfer.ssl.feature</module>
+
+    <module>releng/org.eclipse.ecf.releng.target</module>
+    <module>releng/org.eclipse.ecf.releng.repository</module>
+  </modules>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.eclipse.tycho</groupId>
+        <artifactId>tycho-maven-plugin</artifactId>
+        <version>${tycho-version}</version>
+        <extensions>true</extensions>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.tycho</groupId>
+        <artifactId>target-platform-configuration</artifactId>
+        <version>${tycho-version}</version>
+        <configuration>
+          <target>
+            <artifact>
+              <groupId>org.eclipse.ecf</groupId>
+              <artifactId>org.eclipse.ecf.releng.target</artifactId>
+              <classifier>ecf-${target-platform}</classifier>
+              <version>1.0.0-SNAPSHOT</version>
+            </artifact>
+          </target>
+          <environments>
+            <environment>
+              <os>win32</os>
+              <ws>win32</ws>
+              <arch>x86</arch>
+            </environment>
+            <environment>
+              <os>win32</os>
+              <ws>win32</ws>
+              <arch>x86_64</arch>
+            </environment>
+            <environment>
+              <os>linux</os>
+              <ws>gtk</ws>
+              <arch>x86</arch>
+            </environment>
+            <environment>
+              <os>linux</os>
+              <ws>gtk</ws>
+              <arch>x86_64</arch>
+            </environment>
+            <environment>
+              <os>macosx</os>
+              <ws>cocoa</ws>
+              <arch>x86_64</arch>
+            </environment>
+          </environments>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.tycho</groupId>
+        <artifactId>tycho-source-plugin</artifactId>
+        <version>${tycho-version}</version>
+        <executions>
+          <execution>
+            <id>plugin-source</id>
+            <goals>
+              <goal>plugin-source</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.eclipse.tycho.extras</groupId>
+        <artifactId>tycho-source-feature-plugin</artifactId>
+        <version>${tycho-version}</version>
+        <configuration>
+          <includeBinaryFeature>false</includeBinaryFeature>
+        </configuration>
+        <executions>
+          <execution>
+            <id>source-feature</id>
+            <phase>package</phase>
+            <goals>
+              <goal>source-feature</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.eclipse.tycho</groupId>
+          <artifactId>tycho-compiler-plugin</artifactId>
+          <version>${tycho-version}</version>
+          <configuration>
+            <compilerArguments>
+              <properties>${project.basedir}/.settings/org.eclipse.jdt.core.prefs</properties>
+            </compilerArguments>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.eclipse.tycho</groupId>
+          <artifactId>tycho-packaging-plugin</artifactId>
+          <version>${tycho-version}</version>
+          <dependencies>
+            <dependency>
+              <groupId>org.eclipse.tycho.extras</groupId>
+              <artifactId>tycho-buildtimestamp-jgit</artifactId>
+              <version>${tycho-version}</version>
+            </dependency>
+            <dependency>
+              <groupId>org.eclipse.tycho.extras</groupId>
+              <artifactId>tycho-sourceref-jgit</artifactId>
+              <version>${tycho-version}</version>
+            </dependency>
+          </dependencies>
+          <configuration>
+            <format>'v'yyyyMMdd-HHmm</format>
+            <archive>
+              <addMavenDescriptor>true</addMavenDescriptor>
+            </archive>
+            <sourceReferences>
+              <generate>true</generate>
+            </sourceReferences>
+            <timestampProvider>jgit</timestampProvider>
+            <jgit.dirtyWorkingTree>warning</jgit.dirtyWorkingTree>
+          </configuration>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+  </build>
+
+  <profiles>
+    <profile>
+      <!-- Enable this profile to pack200 and sign p2 repository contents -->
+      <id>pack-and-sign</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.eclipse.tycho</groupId>
+            <artifactId>target-platform-configuration</artifactId>
+            <version>${tycho-version}</version>
+            <configuration>
+              <includePackedArtifacts>true</includePackedArtifacts>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.eclipse.tycho.extras</groupId>
+            <artifactId>tycho-pack200a-plugin</artifactId>
+            <version>${tycho-version}</version>
+            <executions>
+              <execution>
+                <id>pack200-normalize</id>
+                <goals>
+                  <goal>normalize</goal>
+                </goals>
+                <phase>verify</phase>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.eclipse.cbi.maven.plugins</groupId>
+            <artifactId>eclipse-jarsigner-plugin</artifactId>
+            <version>${cbi-version}</version>
+            <executions>
+              <execution>
+                <id>sign</id>
+                <phase>verify</phase>
+                <goals>
+                  <goal>sign</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.eclipse.tycho.extras</groupId>
+            <artifactId>tycho-pack200b-plugin</artifactId>
+            <version>${tycho-version}</version>
+            <executions>
+              <execution>
+                <id>pack200-pack</id>
+                <goals>
+                  <goal>pack</goal>
+                </goals>
+                <phase>verify</phase>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <groupId>org.eclipse.tycho</groupId>
+            <artifactId>tycho-p2-plugin</artifactId>
+            <version>${tycho-version}</version>
+            <executions>
+              <execution>
+                <id>p2-metadata</id>
+                <goals>
+                  <goal>p2-metadata</goal>
+                </goals>
+                <phase>verify</phase>
+              </execution>
+            </executions>
+            <configuration>
+              <defaultP2Metadata>false</defaultP2Metadata>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+  </profiles>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl/pom.xml b/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl/pom.xml
new file mode 100644
index 0000000..6ff8456
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.filetransfer.httpclient4.ssl</artifactId>
+  <version>1.1.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4/pom.xml b/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4/pom.xml
new file mode 100644
index 0000000..77b917c
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.filetransfer.httpclient4</artifactId>
+  <version>1.1.100-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl/pom.xml b/providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl/pom.xml
new file mode 100644
index 0000000..e54ae2c
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.filetransfer.ssl</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.filetransfer/pom.xml b/providers/bundles/org.eclipse.ecf.provider.filetransfer/pom.xml
new file mode 100644
index 0000000..c8f33ff
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.filetransfer/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.filetransfer</artifactId>
+  <version>3.2.200-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.core.feature/pom.xml b/releng/features/org.eclipse.ecf.core.feature/pom.xml
new file mode 100644
index 0000000..e38669d
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.core.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.core.feature</artifactId>
+  <version>1.3.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.core.ssl.feature/pom.xml b/releng/features/org.eclipse.ecf.core.ssl.feature/pom.xml
new file mode 100644
index 0000000..3064359
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.core.ssl.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.core.ssl.feature</artifactId>
+  <version>1.1.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.filetransfer.feature/pom.xml b/releng/features/org.eclipse.ecf.filetransfer.feature/pom.xml
new file mode 100644
index 0000000..5a3ea51
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.filetransfer.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.filetransfer.feature</artifactId>
+  <version>3.13.2-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature/pom.xml b/releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature/pom.xml
new file mode 100644
index 0000000..a85eb0c
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.filetransfer.httpclient4.feature</artifactId>
+  <version>3.13.2-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature/pom.xml b/releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature/pom.xml
new file mode 100644
index 0000000..a7c4a44
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.filetransfer.httpclient4.ssl.feature</artifactId>
+  <version>1.1.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.filetransfer.ssl.feature/pom.xml b/releng/features/org.eclipse.ecf.filetransfer.ssl.feature/pom.xml
new file mode 100644
index 0000000..0cd677a
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.filetransfer.ssl.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.filetransfer.ssl.feature</artifactId>
+  <version>1.1.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/org.eclipse.ecf.releng.repository/.project b/releng/org.eclipse.ecf.releng.repository/.project
new file mode 100644
index 0000000..aab39c5
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.repository/.project
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.eclipse.ecf.releng.repository</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
diff --git a/releng/org.eclipse.ecf.releng.repository/category.xml b/releng/org.eclipse.ecf.releng.repository/category.xml
new file mode 100644
index 0000000..167336c
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.repository/category.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<site>
+   <feature id="org.eclipse.ecf.core.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <feature id="org.eclipse.ecf.core.ssl.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <feature id="org.eclipse.ecf.filetransfer.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <feature id="org.eclipse.ecf.filetransfer.httpclient4.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <feature id="org.eclipse.ecf.filetransfer.httpclient4.ssl.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <feature id="org.eclipse.ecf.filetransfer.ssl.feature" version="0.0.0">
+      <category name="core"/>
+   </feature>
+   <category-def name="core" label="ECF Core Features"/>
+</site>
diff --git a/releng/org.eclipse.ecf.releng.repository/pom.xml b/releng/org.eclipse.ecf.releng.repository/pom.xml
new file mode 100644
index 0000000..0730895
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.repository/pom.xml
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.releng.repository</artifactId>
+  <packaging>eclipse-repository</packaging>
+
+  <name>ECF p2 Repository</name>
+</project>
diff --git a/releng/org.eclipse.ecf.releng.target/.project b/releng/org.eclipse.ecf.releng.target/.project
new file mode 100644
index 0000000..cc61b55
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.target/.project
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>org.eclipse.ecf.releng.target</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
diff --git a/releng/org.eclipse.ecf.releng.target/ecf-luna.target b/releng/org.eclipse.ecf.releng.target/ecf-luna.target
new file mode 100644
index 0000000..99586c4
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.target/ecf-luna.target
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?><target name="Luna" sequenceNumber="19">
+<locations>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.eclipse.pde.feature.group" version="3.10.1.v20150204-1700"/>
+<unit id="org.eclipse.emf.edit.ui.feature.group" version="2.10.2.v20150123-0452"/>
+<unit id="org.eclipse.platform.source.feature.group" version="4.4.2.v20150204-1700"/>
+<unit id="org.eclipse.platform.feature.group" version="4.4.2.v20150204-1700"/>
+<unit id="org.eclipse.pde.source.feature.group" version="3.10.1.v20150204-1700"/>
+<unit id="org.eclipse.emf.edit.ui.source.feature.group" version="2.10.2.v20150123-0452"/>
+<repository location="http://download.eclipse.org/releases/luna"/>
+</location>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.apache.httpcomponents.httpcore" version="4.3.3.v201411290715"/>
+<unit id="org.apache.httpcomponents.httpclient" version="4.3.6.v201411290715"/>
+<unit id="org.jdom.source" version="1.1.1.v201101151400"/>
+<unit id="org.apache.ws.commons.util" version="1.0.1.v20100518-1140"/>
+<unit id="org.apache.xmlrpc" version="3.0.0.v20100427-1100"/>
+<unit id="org.apache.commons.codec.source" version="1.6.0.v201305230611"/>
+<unit id="org.apache.httpcomponents.httpcore.source" version="4.3.3.v201411290715"/>
+<unit id="org.json" version="1.0.0.v201011060100"/>
+<unit id="com.sun.syndication.source" version="0.9.0.v200803061811"/>
+<unit id="org.jdom" version="1.1.1.v201101151400"/>
+<unit id="org.apache.httpcomponents.httpclient.source" version="4.3.6.v201411290715"/>
+<unit id="com.sun.syndication" version="0.9.0.v200803061811"/>
+<unit id="org.apache.log4j.source" version="1.2.15.v201012070815"/>
+<unit id="org.apache.commons.httpclient.source" version="3.1.0.v201012070820"/>
+<unit id="org.apache.commons.httpclient" version="3.1.0.v201012070820"/>
+<unit id="org.apache.hadoop.zookeeper.source" version="3.3.3.v201105210832"/>
+<unit id="org.apache.commons.codec" version="1.6.0.v201305230611"/>
+<unit id="org.apache.hadoop.zookeeper" version="3.3.3.v201105210832"/>
+<unit id="org.xbill.dns.source" version="2.0.8.v201112050911"/>
+<unit id="org.xbill.dns" version="2.0.8.v201112050911"/>
+<unit id="org.apache.log4j" version="1.2.15.v201012070815"/>
+<repository location="http://download.eclipse.org/tools/orbit/downloads/drops/R20150124073747/repository/"/>
+</location>
+</locations>
+</target>
diff --git a/releng/org.eclipse.ecf.releng.target/ecf-mars.target b/releng/org.eclipse.ecf.releng.target/ecf-mars.target
new file mode 100644
index 0000000..0a5d678
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.target/ecf-mars.target
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?><target name="Mars" sequenceNumber="4">
+<locations>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.apache.httpcomponents.httpcore" version="4.3.3.v201411290715"/>
+<unit id="org.jdom.source" version="1.1.1.v201101151400"/>
+<unit id="org.apache.ws.commons.util" version="1.0.1.v20100518-1140"/>
+<unit id="org.apache.xmlrpc" version="3.0.0.v20100427-1100"/>
+<unit id="org.apache.commons.codec.source" version="1.6.0.v201305230611"/>
+<unit id="org.apache.httpcomponents.httpcore.source" version="4.3.3.v201411290715"/>
+<unit id="org.json" version="1.0.0.v201011060100"/>
+<unit id="com.sun.syndication.source" version="0.9.0.v200803061811"/>
+<unit id="org.jdom" version="1.1.1.v201101151400"/>
+<unit id="org.apache.httpcomponents.httpclient" version="4.3.6.v201411290715B"/>
+<unit id="com.sun.syndication" version="0.9.0.v200803061811"/>
+<unit id="org.apache.log4j.source" version="1.2.15.v201012070815"/>
+<unit id="org.apache.commons.httpclient.source" version="3.1.0.v201012070820"/>
+<unit id="org.apache.commons.httpclient" version="3.1.0.v201012070820"/>
+<unit id="org.apache.hadoop.zookeeper.source" version="3.3.3.v201105210832"/>
+<unit id="org.apache.commons.codec" version="1.6.0.v201305230611"/>
+<unit id="org.apache.httpcomponents.httpclient.source" version="4.3.6.v201411290715B"/>
+<unit id="org.apache.hadoop.zookeeper" version="3.3.3.v201105210832"/>
+<unit id="org.xbill.dns.source" version="2.0.8.v201112050911"/>
+<unit id="org.xbill.dns" version="2.0.8.v201112050911"/>
+<unit id="org.apache.log4j" version="1.2.15.v201012070815"/>
+<repository location="http://download.eclipse.org/tools/orbit/downloads/drops/R20160221192158/repository/"/>
+</location>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.eclipse.platform.source.feature.group" version="4.5.2.v20160212-1500"/>
+<unit id="org.eclipse.emf.edit.ui.source.feature.group" version="2.11.1.v20160208-0841"/>
+<unit id="org.eclipse.platform.feature.group" version="4.5.2.v20160212-1500"/>
+<unit id="org.eclipse.pde.source.feature.group" version="3.11.2.v20160212-1500"/>
+<unit id="org.eclipse.pde.feature.group" version="3.11.2.v20160212-1500"/>
+<unit id="org.eclipse.emf.edit.ui.feature.group" version="2.11.1.v20160208-0841"/>
+<repository location="http://download.eclipse.org/releases/mars"/>
+</location>
+</locations>
+</target>
diff --git a/releng/org.eclipse.ecf.releng.target/ecf-neon.target b/releng/org.eclipse.ecf.releng.target/ecf-neon.target
new file mode 100644
index 0000000..cd2924d
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.target/ecf-neon.target
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<?pde version="3.8"?><target name="Neon" sequenceNumber="8">
+<locations>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.apache.httpcomponents.httpcore" version="4.3.3.v201411290715"/>
+<unit id="org.jdom.source" version="1.1.1.v201101151400"/>
+<unit id="org.apache.ws.commons.util" version="1.0.1.v20100518-1140"/>
+<unit id="org.apache.xmlrpc" version="3.0.0.v20100427-1100"/>
+<unit id="org.apache.commons.codec.source" version="1.6.0.v201305230611"/>
+<unit id="org.apache.httpcomponents.httpcore.source" version="4.3.3.v201411290715"/>
+<unit id="org.json" version="1.0.0.v201011060100"/>
+<unit id="com.sun.syndication.source" version="0.9.0.v200803061811"/>
+<unit id="org.jdom" version="1.1.1.v201101151400"/>
+<unit id="org.apache.httpcomponents.httpclient" version="4.3.6.v201511171540"/>
+<unit id="org.apache.httpcomponents.httpclient.source" version="4.3.6.v201511171540"/>
+<unit id="com.sun.syndication" version="0.9.0.v200803061811"/>
+<unit id="org.apache.log4j.source" version="1.2.15.v201012070815"/>
+<unit id="org.apache.commons.httpclient.source" version="3.1.0.v201012070820"/>
+<unit id="org.apache.commons.httpclient" version="3.1.0.v201012070820"/>
+<unit id="org.apache.commons.codec" version="1.6.0.v201305230611"/>
+<unit id="org.xbill.dns.source" version="2.0.8.v201112050911"/>
+<unit id="org.xbill.dns" version="2.0.8.v201112050911"/>
+<unit id="org.apache.log4j" version="1.2.15.v201012070815"/>
+<repository location="http://download.eclipse.org/tools/orbit/downloads/drops/R20160520211859/repository/"/>
+</location>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.eclipse.pde.feature.group" version="3.12.0.v20160606-1100"/>
+<unit id="org.eclipse.emf.edit.ui.source.feature.group" version="2.12.0.v20160526-0356"/>
+<unit id="org.eclipse.pde.source.feature.group" version="3.12.0.v20160606-1100"/>
+<unit id="org.eclipse.emf.edit.ui.feature.group" version="2.12.0.v20160526-0356"/>
+<unit id="org.eclipse.platform.feature.group" version="4.6.0.v20160606-1342"/>
+<unit id="org.eclipse.platform.source.feature.group" version="4.6.0.v20160606-1342"/>
+<repository location="http://download.eclipse.org/releases/neon"/>
+</location>
+<location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.apache.hadoop.zookeeper" version="3.3.3.v201105210832"/>
+<unit id="org.apache.hadoop.zookeeper.source" version="3.3.3.v201105210832"/>
+<repository location="http://download.eclipse.org/tools/orbit/downloads/drops/R20160221192158/repository/"/>
+</location>
+</locations>
+</target>
diff --git a/releng/org.eclipse.ecf.releng.target/pom.xml b/releng/org.eclipse.ecf.releng.target/pom.xml
new file mode 100644
index 0000000..bca1f3c
--- /dev/null
+++ b/releng/org.eclipse.ecf.releng.target/pom.xml
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.releng.target</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>pom</packaging>
+
+  <name>ECF Target Platform Definitions</name>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>1.10</version>
+        <executions>
+          <execution>
+            <id>attach-artifacts</id>
+            <phase>package</phase>
+            <goals>
+              <goal>attach-artifact</goal>
+            </goals>
+            <configuration>
+              <artifacts>
+                <artifact>
+                  <file>ecf-luna.target</file>
+                  <type>target</type>
+                  <classifier>ecf-luna</classifier>
+                </artifact>
+                <artifact>
+                  <file>ecf-mars.target</file>
+                  <type>target</type>
+                  <classifier>ecf-mars</classifier>
+                </artifact>
+                <artifact>
+                  <file>ecf-neon.target</file>
+                  <type>target</type>
+                  <classifier>ecf-neon</classifier>
+                </artifact>
+              </artifacts>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
+</project>
diff --git a/framework/bundles/org.eclipse.ecf.discovery/pom.xml b/framework/bundles/org.eclipse.ecf.discovery/pom.xml
new file mode 100644
index 0000000..39abef6
--- /dev/null
+++ b/framework/bundles/org.eclipse.ecf.discovery/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.discovery</artifactId>
+  <version>5.0.200-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/pom.xml b/pom.xml
index b2470aa..a6a692c 100644
--- a/pom.xml
+++ b/pom.xml
@@ -67,11 +67,16 @@
     <module>framework/bundles/org.eclipse.ecf.filetransfer</module>
     <module>framework/bundles/org.eclipse.ecf.identity</module>
     <module>framework/bundles/org.eclipse.ecf.ssl</module>
+    <module>framework/bundles/org.eclipse.ecf.discovery</module>
 
     <module>providers/bundles/org.eclipse.ecf.provider.filetransfer</module>
     <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4</module>
     <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.httpclient4.ssl</module>
     <module>providers/bundles/org.eclipse.ecf.provider.filetransfer.ssl</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.discovery</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.dnssd</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.jmdns</module>
+    <module>providers/bundles/org.eclipse.ecf.provider.zookeeper</module>
 
     <module>releng/features/org.eclipse.ecf.core.feature</module>
     <module>releng/features/org.eclipse.ecf.core.ssl.feature</module>
@@ -79,6 +84,10 @@
     <module>releng/features/org.eclipse.ecf.filetransfer.httpclient4.feature</module>
     <module>releng/features/org.eclipse.ecf.filetransfer.httpclient4.ssl.feature</module>
     <module>releng/features/org.eclipse.ecf.filetransfer.ssl.feature</module>
+    <module>releng/features/org.eclipse.ecf.discovery.feature</module>
+    <module>releng/features/org.eclipse.ecf.discovery.dnssd.feature</module>
+    <module>releng/features/org.eclipse.ecf.discovery.jmdns.feature</module>
+    <module>releng/features/org.eclipse.ecf.discovery.zookeeper.feature</module>
 
     <module>releng/org.eclipse.ecf.releng.target</module>
     <module>releng/org.eclipse.ecf.releng.repository</module>
diff --git a/providers/bundles/org.eclipse.ecf.provider.discovery/pom.xml b/providers/bundles/org.eclipse.ecf.provider.discovery/pom.xml
new file mode 100644
index 0000000..1a77273
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.discovery/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.discovery</artifactId>
+  <version>3.0.0-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.dnssd/pom.xml b/providers/bundles/org.eclipse.ecf.provider.dnssd/pom.xml
new file mode 100644
index 0000000..3802da6
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.dnssd/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.dnssd</artifactId>
+  <version>1.2.100-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.jmdns/pom.xml b/providers/bundles/org.eclipse.ecf.provider.jmdns/pom.xml
new file mode 100644
index 0000000..b9e3102
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.jmdns/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.jmdns</artifactId>
+  <version>4.3.100-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/providers/bundles/org.eclipse.ecf.provider.zookeeper/pom.xml b/providers/bundles/org.eclipse.ecf.provider.zookeeper/pom.xml
new file mode 100644
index 0000000..aa50958
--- /dev/null
+++ b/providers/bundles/org.eclipse.ecf.provider.zookeeper/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.provider.zookeeper</artifactId>
+  <version>1.2.100-SNAPSHOT</version>
+  <packaging>eclipse-plugin</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.discovery.dnssd.feature/pom.xml b/releng/features/org.eclipse.ecf.discovery.dnssd.feature/pom.xml
new file mode 100644
index 0000000..8c4a5a1
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.discovery.dnssd.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.discovery.dnssd.feature</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.discovery.feature/pom.xml b/releng/features/org.eclipse.ecf.discovery.feature/pom.xml
new file mode 100644
index 0000000..0329c3f
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.discovery.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.discovery.feature</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.discovery.jmdns.feature/pom.xml b/releng/features/org.eclipse.ecf.discovery.jmdns.feature/pom.xml
new file mode 100644
index 0000000..d9f97d0
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.discovery.jmdns.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.discovery.jmdns.feature</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
diff --git a/releng/features/org.eclipse.ecf.discovery.zookeeper.feature/pom.xml b/releng/features/org.eclipse.ecf.discovery.zookeeper.feature/pom.xml
new file mode 100644
index 0000000..7a5acb2
--- /dev/null
+++ b/releng/features/org.eclipse.ecf.discovery.zookeeper.feature/pom.xml
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.eclipse.ecf</groupId>
+    <artifactId>ecf-parent</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+    <relativePath>../../../</relativePath>
+  </parent>
+  <groupId>org.eclipse.ecf</groupId>
+  <artifactId>org.eclipse.ecf.discovery.zookeeper.feature</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>eclipse-feature</packaging>
+</project>
-- 
cgit v0.11.2-4-g4a35


